### PR TITLE
Session cohort

### PR DESCRIPTION
**Pull request by misaka**
There are a few (known) issues that should be addressed in follow-up PRs:
- The spacing doesn't quite match the prototype, in particular the space between the checkbox and the patient name is considerably bigger.
- There's no checkbox at the top to select/deselect all patients.
- The validations may need a little bit of tweaking, not sure if there should be a red line next to all the checkboxes for example, or whether the linking should work.

Also, the session create tests are getting pretty big. I'd propose we move the validations to a separate test file.

image

## Validations

image

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -505,3 +526,5 @@ en:
             time_of_day:
               blank: Select a time of day
               inclusion: Select a time of day
+            patients:
+              blank: Cannot continue without selecting patients
```
Comment (tvararu) [inline config/locales/en.yml:530]:
```suggestion
              blank: Select one or more patients to add to the session
```

File: app/models/patient.rb [hunk 2]
```diff
@@ -70,6 +70,13 @@ def vaccination_records_for_session(session)
     patient_sessions.find_by_session_id(session.id).vaccination_records
   end
 
+  def year_group
+    first_school_year = date_of_birth.year + 5
+    first_school_year += 1 if date_of_birth.month >= 9
+    Time.zone.today.year - first_school_year +
+      (Time.zone.today.month >= 9 ? 1 : 0)
+  end
```
Comment (tvararu) [inline app/models/patient.rb:78]: I get the feeling this can be golfed but 🚢

**Review by tvararu — APPROVED** [review]

File: app/models/patient.rb (hunk 2 above)
Comment (tvararu) [reply app/models/patient.rb:78]: immediately and then applies the two conditional increments:

```suggestion
  def year_group
    result = Time.zone.today.year - date_of_birth.year - 5
    result -= 1 if date_of_birth.month >= 9
    result += 1 if Time.zone.today.month >= 9
    result
  end
```

Accept it if you think it's clearer, reject if not 👍

Comment (misaka) [reply app/models/patient.rb:78]: Ya, 🚢 for now I say. Once we have the baseline set we can start playing golf with it. 😄

File: app/controllers/edit_sessions_controller.rb [hunk 3]
```diff
@@ -88,6 +88,7 @@ def update_params
 
   def set_steps
     self.steps = @session.form_steps
+    @previous_step = previous_step
```
[the remark this answers is not in the document]
Comment (misaka) [reply app/controllers/edit_sessions_controller.rb:96]: I don't believe this is available in the views. Unless I was doing something wrong ... but I did try it and got an error, and it appears to be documented as a controller helper (vs views helpers) here: https://github.com/zombocom/wicked#quick-reference

File: config/locales/en.yml (hunk 1 above)
Comment (misaka) [reply config/locales/en.yml]: Fix from a mini snag session with Emma this morning. 🙏

**Pull request merged**